price_pusher/src/near/near.ts

```diff
@@ -0,0 +1,194 @@
+      return;
+    }
+
+    console.log("Pushing ", priceIds);
+
+    for (const data of priceFeedUpdateData) {
+      let updateFee;
+      try {
+        updateFee = await this.account.getUpdateFeeEstimate(data);
+        console.log(`Update fee: ${updateFee}`);
+      } catch (e: any) {
+        console.error(new Date(), "getUpdateFeeEstimate failed:", e);
+        continue;
+      }
+
+      try {
+        const outcome = await this.account.updatePriceFeeds(data, updateFee);
+        const failureMessages: (ExecutionStatus | ExecutionStatusBasic)[] = [];
+        const is_success = Object.values(outcome['receipts_outcome']).reduce((is_success, receipt) => {
+          if (Object.prototype.hasOwnProperty.call(receipt["outcome"]["status"], "Failure")) {
+            failureMessages.push(receipt["outcome"]["status"])
+            return false;
+          }
+          return is_success;
+        }, true);
+        if (is_success) {
+          console.log(new Date(), "updatePriceFeeds successful. Tx hash: ", outcome["transaction"]["hash"]);
+        } else {
+          console.error(new Date(), "updatePriceFeeds failed:", JSON.stringify(failureMessages, undefined, 2));
+        }
+      } catch (e: any) {
+        console.error(new Date(), "updatePriceFeeds failed:", e);
+      }
+    }
+  }
+
+  private async getPriceFeedsUpdateData(
+    priceIds: HexString[]
+  ): Promise<string[]> {
+    const latestVaas = await this.connection.getLatestVaas(priceIds);
+    return latestVaas.map(
+      (vaa) => Buffer.from(vaa, "base64").toString("hex")
+    );
+  }
+}
+
+export class NearAccount {
+  private account: Account;
+
+  constructor(
+    network: string,
+    accountId: string,
+    nodeUrl: string,
+    private pythAccountId: string
+  ) {
+    const connection = this.getConnection(network, accountId, nodeUrl);
+    this.account = new Account(connection, accountId);
+  }
+
+  async getPriceUnsafe(priceId: string): Promise<any> {
+    return await this.account.viewFunction({
+      contractId: this.pythAccountId,
+      methodName: "get_price_unsafe",
+      args: {
+        price_identifier: priceId
+      }
+    });
+  }
+
+  async getUpdateFeeEstimate(data: string): Promise<any> {
+    return await this.account.viewFunction({
+      contractId: this.pythAccountId,
+      methodName: "get_update_fee_estimate",
+      args: {
+        data
+      }
+    });
+  }
+
+  async updatePriceFeeds(data: string, updateFee: any): Promise<FinalExecutionOutcome> {
+    return await this.account.functionCall({
+      contractId: this.pythAccountId,
+      methodName: "update_price_feeds",
+      args: {
+        data
+      },
+      gas: "300000000000000" as any,
+      attachedDeposit: updateFee
+    });
+  }
+
+  private getConnection(network: string, accountId: string, nodeUrl: string): Connection {
+    const content = fs.readFileSync(path.join(os.homedir(), ".near-credentials", network, accountId + ".json"));
```

Comment:
This is not very good. Could you get the path from the arguments like other pushers and use that?

Reply:
Okay, added the optional parameter 'private-key-path'.

Reply:
This path is where NEAR by default stores private keys, and users are highly likely to place their private keys here. Additionally, the official near-cli also reads private keys from this location. Therefore, the parameter for specifying the private key file path has been set as optional.